src/presenter/filter-presenter.js

```diff
@@ -1,57 +1,189 @@
-import { render } from '../framework/render';
+import { render, remove } from '../framework/render';
 
 import FilterCategoryView from '../view/filter-category-view';
 import FilterTimeView from '../view/filter-time-view';
-import AddNewTripButtonView from '../view/add-new-trip-button-view';
 import InfoView from '../view/info-view';
 
-import {FilterType , SortType} from '../const.js';
+
+import { FilterType, SortType, UpdateType} from '../const.js';
+import { filterListFuture, filterListPresent, filterListPast, sortListDay, sortListPrice, sortListTime } from '../utils';
 
 
 export default class FilterPresenter{
   #tripFilterCategoryContainer = null;
   #tripHeaderContainer = null;
-  #tripPointBoardPresenter = null;
+  #tripFilterContainer = null;
+
 
-  #currentFilterType = FilterType.EVERYTHING;
-  #currentSortCategory = SortType.DAY;
+  #filterModel = null;
+  #tripPointsModel = null;
+  #sortModel = null;
+
+  #filterTimeViewComponent = null;
+  #filterCategoryViewComponent = null;
 
   constructor(
     {
       tripFilterCategoryContainer,
       tripHeaderContainer,
-      tripPointBoardPresenter
+      tripFilterContainer,
+
+      filterModel,
+      tripPointsModel,
+      sortModel
     }
   ){
+
     this.#tripFilterCategoryContainer = tripFilterCategoryContainer;
     this.#tripHeaderContainer = tripHeaderContainer;
-    this.#tripPointBoardPresenter = tripPointBoardPresenter;
+    this.#tripFilterContainer = tripFilterContainer;
+
+    this.#filterModel = filterModel;
+    this.#tripPointsModel = tripPointsModel;
+    this.#sortModel = sortModel;
+
+    this.#tripPointsModel.addObserver(this.#setSortFilterDefault);
+    this.#tripPointsModel.addObserver(this.#updateFilterSortView);
+
+    this.#filterModel.addObserver(this.#updateSoortView);
+
   }
 
   init() {
-    render(new FilterCategoryView({ handleSortCategoryChange: this.#handleSortCategoryChange}), this.#tripFilterCategoryContainer);
-    render(new InfoView(), this.#tripHeaderContainer);
-    render(new FilterTimeView({ handleFilterTypeChange:this.#handleFilterTypeChange}), this.#tripHeaderContainer);
-    render(new AddNewTripButtonView(), this.#tripHeaderContainer);
+    render(new InfoView(), this.#tripHeaderContainer, 'afterbegin');
```

Comment:
Вместе с методом `render` есть перечисление для позиции:
```suggestion
    render(new InfoView(), this.#tripHeaderContainer, RenderPosition.AFTERBEGIN);
```